### ISSUE TITLE
UEFIHelper.java script error: java.lang.NullPointerException: Cannot invoke "ghidra.program.model.listing.Function.getParameters()" because "function" is null

### ISSUE THREAD

**Issue by Bohosul01** [al3xtjames/ghidra-firmware-utils#29]
I'm using JDK 21 with Ghidra 10.2.1 and I'm trying to analyze the `bootx64.efi` file from my machine's EFI partition.

When I run the script, it is able to locate the entrypoint, but then it throws the  `java.lang.NullPointerException: Cannot invoke "ghidra.program.model.listing.Function.getParameters()" because "function" is null` exception.

I have also tried to reverse `bootx64.efi` using efiseek, but it won't make any changes. Could this bug happen  due to analyzing a bootloader? If so, how to fix it?

**Comment by al3xtjames** [comment]
Can you upload the bootloader binary (bootx64.efi) and post the full stack trace from the `NullPointerException`?

**Comment by Bohosul01** [comment]
[bootx64.zip](https://github.com/al3xtjames/ghidra-firmware-utils/files/14221661/bootx64.zip)

```
Cannot invoke "ghidra.program.model.listing.Function.getParameters()" because "function" is null
java.lang.NullPointerException: Cannot invoke "ghidra.program.model.listing.Function.getParameters()" because "function" is null
	at UEFIHelper.propagateFunctionParameters(UEFIHelper.java:261)
	at UEFIHelper.propagateFunctionParameters(UEFIHelper.java:307)
	at UEFIHelper.propagateFunctionParameters(UEFIHelper.java:307)
	at UEFIHelper.propagateFunctionParameters(UEFIHelper.java:307)
	at UEFIHelper.defineEntryPoint(UEFIHelper.java:349)
	at UEFIHelper.run(UEFIHelper.java:477)
	at ghidra.app.script.GhidraScript.executeNormal(GhidraScript.java:395)
	at ghidra.app.script.GhidraScript.doExecute(GhidraScript.java:250)
	at ghidra.app.script.GhidraScript.execute(GhidraScript.java:228)
	at ghidra.app.plugin.core.script.RunScriptTask.run(RunScriptTask.java:47)
	at ghidra.util.task.Task.monitoredRun(Task.java:134)
	at ghidra.util.task.TaskRunner.lambda$startTaskThread$0(TaskRunner.java:106)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)

---------------------------------------------------
Build Date: 2022-Nov-10 1636 EST
Ghidra Version: 10.2.1
Java Home: C:\Program Files\Eclipse Adoptium\jdk-21.0.2.13-hotspot
JVM Version: Eclipse Adoptium 21.0.2
OS: Windows 10 10.0 amd64
```

**Comment by al3xtjames** [comment]
Did you let autoanalysis complete before running UEFIHelper? I only saw the `NullPointerException` while running it against an unanalyzed binary.

**Comment by Bohosul01** [comment]
As you mentioned, it works if autoanalysis completes before running the script. I'm new at reverse engineering and I thought that autoanalysis would only recognize some patterns and simplify executable analysis for the user, not that it would be necessary to run scripts.

I found some online tutorials about reverse engineering BIOS modules using Ghidra, but I couldn't find anything about bootloaders, so I imagined that bootloaders were somehow different to other EFI executables and that could be provoking the exception.
